app/controllers/users_controller.rb

```diff
@@ -42,13 +43,24 @@ def destroy
     redirect_to users_url, notice: notice_message
   end
 
+  def edit_password; end
+
+  def update_password
+    if @user.update(user_params)
+      sign_in(@user, bypass: true)
+      redirect_to edit_password_users_url, notice: 'Password was successfully updated.'
+    else
+      render :edit_password
+    end
+  end
+
```

Comment:
I tend to prefer to keep my controllers and routes as RESTful as possible. i.e Only having controllers with the actions that are generated for free from `resources` or `resource` in the routes file. How would you feel about pulling this out into a `PasswordsController#edit` and `PasswordsController#update` action?